### PR TITLE
feat: add fuel wallet

### PR DESCRIPTION
**Pull request by Dhaiwat10**
# Background
I would love to add full Fuel support to the GOAT SDK, and I figured a good place to start would be adding the wallet; followed by plugins later :)

## What does this PR do?
This PR adds a `FuelWalletClient` (`@goat-sdk/wallet-fuel`).

## What kind of change is this?
New wallet

# Documentation changes needed?

Yes. Done

# Testing

## Detailed testing steps
- cd into the `vercel-ai/fuel` example
- Populate the `.env` file with a Fuel Wallet private key and an OpenAI API key
- Run `pnpm send-eth`

## For plugins
- [x] I have tested this change locally with key pair wallets
- [ ] I have tested this change locally with hosted wallets (e.g. Crossmint Smart Wallets, etc.)

## Discord username
dhaiwat10

### CODE REVIEW

File: typescript/packages/wallets/fuel/src/FuelWalletClient.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import { WalletClientBase } from "@goat-sdk/core";
+import { bn, formatUnits, TransactionRequest, WalletUnlocked } from "fuels";
+
+export type FuelWalletCtorParams = {
+    fuelWallet: WalletUnlocked;
+};
+
+export class FuelWalletClient extends WalletClientBase {
```
Comment (0xaguspunk) [inline typescript/packages/wallets/fuel/src/FuelWalletClient.ts:8]: I would make this an abstract class and create an implementation on top of it like with the solana and evm wallets: 
* https://github.com/goat-sdk/goat/blob/main/typescript/packages/wallets/solana/src/SolanaWalletClient.ts
* https://github.com/goat-sdk/goat/blob/main/typescript/packages/wallets/solana/src/SolanaKeypairWalletClient.ts

This allows you to have other wallet implementations in the future (e.g smart wallets, crossmint wallets, etc.) and have all plugins still work for them :)

**Review by 0xaguspunk — COMMENTED** [review]
Awesome!! 🔥 

Made one suggestion on the wallet implementation!

File: typescript/packages/wallets/fuel/src/FuelWalletClient.ts (hunk 1 above)
Comment (Dhaiwat10) [reply typescript/packages/wallets/fuel/src/FuelWalletClient.ts]: Ty!

https://github.com/goat-sdk/goat/pull/126/commits/e8b890bc7256ea904b3ca61477ade58146dcaf7c

File: typescript/examples/vercel-ai/fuel/CHANGELOG.md
Comment (0xaguspunk) [inline typescript/examples/vercel-ai/fuel/CHANGELOG.md:1]: No need for this file :) it will get generated automatically

**Review by 0xaguspunk — APPROVED** [review]

**Comment by 0xaguspunk** [comment]
Need to run `pnpm lint:fix` only

File: typescript/examples/vercel-ai/fuel/CHANGELOG.md
Comment (Dhaiwat10) [reply typescript/examples/vercel-ai/fuel/CHANGELOG.md:1]: https://github.com/goat-sdk/goat/pull/126/commits/e460098ca1b559bfaa93fa348df9e53dcbadbbfc

**Pull request merged**